go/libraries/doltcore/diff/table_diff.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/fatih/color"
+
+	"github.com/liquidata-inc/dolt/go/cmd/dolt/cli"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/row"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/sql"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/table/pipeline"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/table/typed/noms"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/table/untyped/nullprinter"
+	"github.com/liquidata-inc/dolt/go/libraries/utils/iohelp"
+)
+
+type tableDiff struct {
+	adds    []string
+	drops   []string
+	renames map[string]string
+	same    []string
+}
+
+func SQLTableDIffs(ctx context.Context, r1, r2 *doltdb.RootValue) error {
```

Comment:
Would call this PrintSqlTableDiffs, and pass in the CLI as a io.WriteCloser instead of assuming it. Then you can write a unit test of this method to make sure it prints what it should for a few cases